cdk/l3-cdk-resources/atlas-basic/src/index.ts

```diff
@@ -1,98 +1,414 @@
+  /**
+     * @description
+     * @type {user.CfnDatabaseUserPropsX509Type}
+     * @memberof DatabaseUserProps
+     */
+  readonly x509Type?: user.CfnDatabaseUserPropsX509Type;
+  /**
+     * @description
+     * @type {string}
+     * @memberof DatabaseUserProps
+     */
+  readonly password?: string;
+  /**
+     * @description
+     * @type {string}
+     * @memberof DatabaseUserProps
+     */
+  readonly projectId?: string;
+  /**
+     * @description
+     * @type {user.RoleDefinition[]}
+     * @memberof DatabaseUserProps
+     */
+  readonly roles?: user.RoleDefinition[];
+  /**
+     * @description
+     * @type {user.ScopeDefinition[]}
+     * @memberof DatabaseUserProps
+     */
+  readonly scopes?: user.ScopeDefinition[];
+  /**
+     * @description
+     * @type {string}
+     * @memberof DatabaseUserProps
+     */
+  readonly username?: string;
+}
+/**
+ * @description
+ * @export
+ * @interface IpAccessListProps
+ */
+export interface IpAccessListProps {
+  /**
+     * @description
+     * @type {ipAccessList.AccessListDefinition[]}
+     * @memberof IpAccessListProps
+     */
+  readonly accessList: ipAccessList.AccessListDefinition[];
+  /**
+     * @description
+     * @type {string}
+     * @memberof IpAccessListProps
+     */
+  readonly projectId?: string;
+  /**
+     * @description
+     * @type {number}
+     * @memberof IpAccessListProps
+     */
+  readonly totalCount?: number;
+  /**
+     * @description
+     * @type {ipAccessList.ListOptions}
+     * @memberof IpAccessListProps
+     */
+  readonly listOptions?: ipAccessList.ListOptions;
+}
+/**
+ * @description
+ * @export
+ * @interface ApiKeyDefinition
+ */
+export interface ApiKeyDefinition {
+  /**
+     * @description
+     * @type {string}
+     * @memberof ApiKeyDefinition
+     */
+  readonly privateKey?: string;
+  /**
+     * @description
+     * @type {string}
+     * @memberof ApiKeyDefinition
+     */
+  readonly publicKey?: string;
+
+}
+/**
+ * @description
+ * @export
+ * @interface AtlasBasicProps
+ * @extends {cdk.StackProps}
+ */
+export interface AtlasBasicProps extends cdk.StackProps{
+  /**
+     * @description
+     * @type {ApiKeyDefinition}
+     * @memberof AtlasBasicProps
+     */
+  readonly apiKeys : ApiKeyDefinition;
+  /**
+     * @description
+     * @type {ProjectProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly projectProps: ProjectProps;
+  /**
+     * @description
+     * @type {ClusterProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly clusterProps: ClusterProps;
+  /**
+     * @description
+     * @type {DatabaseUserProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly dbUserProps?: DatabaseUserProps;
+  /**
+     * @description
+     * @type {IpAccessListProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly ipAccessListProps?: IpAccessListProps;
+}
+/**
+ * @description
+ * @export
+ * @class AtlasBasic
+ * @extends {cdk.Stack}
+ */
 export class AtlasBasic extends cdk.Stack {
```

Comment:
Can we please include some information/example of how this construct will be used? Or how this was tested, how do the defaults come into play, etc? 
Just trying to get an idea of the UX for this construct.

Reply:
It's available in the API.md. Let me know if it isn't enough.
https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/ebaacac5b31e6204e4d3374ed628ed8db07885df/cdk/l3-cdk-resources/atlas-basic/API.md

Reply:
There is a sample ]([app](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/22b18ad386a65544a29d6b64f689e51d85cd13f6/cdk/l3-cdk-resources/atlas-basic/src)/integ.default.ts) to use this construct. I have updated Readme as well.